crates/subspace-networking/src/behavior/persistent_parameters.rs

```diff
@@ -434,8 +434,8 @@ impl NetworkingParameters {
     }
 }
 
-// Removes a P2p protocol suffix from the multiaddress if any.
-fn remove_p2p_suffix(address: Multiaddr) -> Multiaddr {
+/// Removes a P2p protocol suffix from the multiaddress if any.
+pub(crate) fn remove_p2p_suffix(address: Multiaddr) -> Multiaddr {
     let mut modified_address = address.clone();
```

Comment:
It triggers me a bit that in order to call this function you have cloned an address and then cloned again here and in the end might return an original.

Can you just push back if returned element is not `Protocol::P2p` so you don't need to clone it here?